### DISCUSSION TITLE
Request guard custom errors

### DISCUSSION THREAD

**Discussion by jacks0n9** [rwf2/Rocket#2863] [Questions]
I have a custom error type that just has a message field, and it implements the responder trait, which just serializes it into json. I would like to use this type throughout my entire application, however, for request guards, the Error associated type in the trait isn't actually returned to the user of the api, instead, it's just logged in the console and the catcher for that status code is used instead of my custom error type being returned to the user of the api. is there a way to change this behavior?

Here is my custom error type if it helps:
```rs
#[derive(Serialize, Debug)]
struct Error {
    message: String,
    status: Status,
}
impl<'r, 'o> Responder<'r, 'static> for Error {
    fn respond_to(self, request: &'r rocket::Request<'_>) -> rocket::response::Result<'static> {
        let serialized: String = serde_json::to_string(&self).unwrap();
        Response::build_from(serialized.respond_to(request)?)
            .status(self.status)
            .header(ContentType::JSON)
            .ok()
    }
}
```

**Comment by the10thWiz** [comment]
Sort of, although this will be much easier in 0.6.

Today, you can declare your request guards to be `Result<T, Error>`, which you can then extract via `?` to simply return the error to the user.

I'm currently working on a feature to implement typed errors, which would allow a catcher to directly catch your error struct, and could then respond with it.

**Comment by jacks0n9** [comment]
ok, great. thank you!
